shimming-toolbox/shimmingtoolbox/masking/mask_mrs.py

```diff
@@ -0,0 +1,89 @@
+        raw_data (str): Input path of the siemens raw-data (supported extension .rda)
+        center (list): Voxel's center position in mm of the x, y and z scanner coordinates
+        size (list): Voxel size in mm of the x, y and z scanner coordinates
+    Returns:
+        numpy.ndarray: Cubic mask with the same dimensions as the MRS voxel.
+    """
+
+    if fname_input is None:
+        raise TypeError(f"The input field map needs to be specified")
+
+    if raw_data is None:
+        logger.info("MRS raw data not provided, creating the mask with the given voxel position and size info")
+        if center is None or size is None:
+            raise TypeError('The raw_data is not provided; the voxel position and size are required to proceed')
+        else:
+            mrs_voxel_size = size
+            logger.debug('center:', center)
+            scanner_coordinate = np.array(center + (1, ))
+            logger.debug(f"Scanner position is: {scanner_coordinate}")
+
+    else:
+        logger.info("Reading the raw_data")
+        with tempfile.TemporaryDirectory(prefix='st_' + pathlib.Path(__file__).stem) as tmp:
+            basename = os.path.basename(raw_data)
+            run_subprocess(['spec2nii', 'rda', '-o', tmp, '-f', basename, raw_data])
+            fname_rawdata_nifti = os.path.join(tmp, basename + '.nii.gz')
+            nii = nib.load(fname_rawdata_nifti)
+            header_raw_data = nii.header
+            affine = nii.affine
+        position_sag = header_raw_data['qoffset_x']
+        position_cor = header_raw_data['qoffset_y']
+        position_tra = header_raw_data['qoffset_z']
+        logger.debug(f"raw_data header: {header_raw_data}")
+        logger.debug(f"affine: {affine}")
+        scanner_coordinate = np.array([position_sag, position_cor, position_tra, 1])
+        logger.debug(f"Scanner position is: {scanner_coordinate}")
+        mrs_voxel_size = header_raw_data['pixdim'][1:4]
+
+    logger.debug('mrs_voxel_size is:', mrs_voxel_size)
+    fmap_nii = nib.load(fname_input)
+    fmap_array = fmap_nii.get_fdata()
+    fmap_affine = fmap_nii.affine
+    fmap_header = fmap_nii.header
+    logger.debug('reference_affine:', fmap_affine)
+    logger.debug('reference_affine shape:', np.shape(fmap_affine))
+    voxel_position = npl.inv(fmap_affine).dot(scanner_coordinate)
+    voxel_position = np.round(voxel_position)
+    logger.debug('voxel_position:', voxel_position)
+
+    # The given coordinate (i, j, k) is the voxel's center position.
+    i, j, k = map(int, voxel_position[:3])
+    fmap_resolution = fmap_header['pixdim'][1:4]
+
+    # The distance from the center of the MRS voxel to its edges is calculated based on number of fieldmap voxels.
+    sd = np.ceil(mrs_voxel_size / (2 * fmap_resolution)).astype(int)
+
+    # create a zero mask with the same size as the input fieldmap to be shimmed.
+    mask = np.zeros(fmap_array.shape)
```

Comment:
no header for the mask?

Reply:
I'm not sure what you mean? What header?

Reply:
Right now it seems the mask is only a numpy array and it allows the mask creation without a nifti as input. Wouldn't it be a problem if there is no affine transform?

Reply:
The conversion happens in the CLI